frontend/src/metabase/account/profile/components/UserProfileForm/UserProfileForm.jsx

```diff
@@ -1,22 +1,24 @@
 import React, { useCallback } from "react";
 import PropTypes from "prop-types";
+
 import User from "metabase/entities/users";
+import { usePrevious } from "metabase/hooks/use-previous";
 
 const propTypes = {
   user: PropTypes.object,
 };
 
 const UserProfileForm = ({ user }) => {
-  const handleSaved = useCallback(
-    ({ locale }) => {
-      if (locale !== user.locale) {
-        window.location.reload();
-      }
-    },
-    [user],
-  );
+  const previousUser = usePrevious(user);
```

Comment:
This must be a regression for quite some time. I log both updated user that's passed via `handleSaved` and `user` from this component's prop and their `locale` are the same every time.

I log and found out redux has updated all the states and passed new `user` even before calling the `handleSaved`, so that's why I need to compare the locale with previous recorded user here.